### PR TITLE
Fix/login 401 bug

### PR DESCRIPTION
**Pull request by William-Unosquare**
- Fixes issue where our Axios instance attempted to retrieve the token from the localStorage once, leading it to stay undefined if the user wasn't logged in when this was attempted. There's now an interceptor (think middleware for our requests) that re-retrieves the token.

- A bunch of code refactoring around how we got the user's info. Namely, it all comes from the initial dashboard `getUserInfo` call, instead of bits being hacked together from the email. The `getUserInfo` request now uses an ID call instead of getUserByFirstnameAndLastname.

### CODE REVIEW

File: web-app/src/components/common/Errorbox/styled.js [hunk 1]
```diff
@@ -1,18 +1,25 @@
 import styled from 'styled-components';
 
 export const ErrorContainer = styled.div`
+<<<<<<< HEAD
```
Comment (bgribz) [inline web-app/src/components/common/Errorbox/styled.js]: Missed a wee conflict here

File: web-app/src/hoc/withAuth.js [hunk 2]
```diff
@@ -36,12 +32,7 @@ const withAuth = AuthComponent => {
         return <Redirect to="/login" />;
       }
       if (this.state.user) {
-        return (
-          <AuthComponent
-            user={{ ...this.state.user, ...UserInfo }}
-            {...this.props}
-          />
-        );
+        return <AuthComponent user={{ ...this.state.user }} {...this.props} />;
```
Comment (bgribz) [inline web-app/src/hoc/withAuth.js]: Can this not just be `user={this.state.user}`? Or even `{...this.state}`

**Review by bgribz — CHANGES_REQUESTED** [review]
Just a few things I noticed looking over this

File: web-app/src/components/common/Errorbox/styled.js (hunk 1 above)
Comment (William-Unosquare) [reply web-app/src/components/common/Errorbox/styled.js]: Good spot... Don't know why this hasn't been picked up by my editor or Github 😕

**Review by bgribz — APPROVED** [review]

**Pull request merged**